### DISCUSSION TITLE
Shouldn't commands_post run after a SIGINT?

### DISCUSSION THREAD

**Discussion by ssbarnea** [tox-dev/tox#2377] [Q&A]
If a SIGINT (Ctrl-C) happens during test run, `commands_post` is not run, so basically we have no chance of doing any cleaning.

IMHO, I think that `commands_post` should run instead of just existing quickly.

**Comment by gaborbernat** [comment]
> If a SIGINT (Ctrl-C) happens during test run, `commands_post` is not run, so basically we have no chance of doing any cleaning.

This is by design, and I don't see how we can change this. We can add a flag that would enable users to opt-in for a command post to always run.

**Reply by ssbarnea** [reply]
I can understand that changing this would be really problematic, but a general opt-in to make it always run would seem ok to me. Basically what I described like `if: always()` from github actions (hopefully we would not endup with the same unfortunate naming).

**Comment by carlcsaposs-canonical** [comment]
Ran into this as well—it would be useful to have a `commands_post` or something similar that always runs if `commands_pre` run

Or maybe something like first Ctrl-C interrupts `commands` and second Ctrl-C interrupts `commands_post`—but I imagine that might break uses of the current behavior

**Comment by gaborbernat** [comment]
Converted to issue https://github.com/tox-dev/tox/issues/3858
